contrib/runners/orquesta_runner/tests/unit/test_notify.py

```diff
@@ -0,0 +1,261 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+
+import mock
+
+from orquesta import states as wf_states
+
+import st2tests
+
+# XXX: actionsensor import depends on config being setup.
+import st2tests.config as tests_config
+tests_config.parse_args()
+
+from tests.unit import base
+
+from st2actions.notifier import notifier
+from st2common.bootstrap import actionsregistrar
+from st2common.bootstrap import runnersregistrar
+from st2common.constants import action as action_constants
+from st2common.models.api import notification as notify_api_models
+from st2common.models.db import liveaction as lv_db_models
+from st2common.persistence import execution as ex_db_access
+from st2common.persistence import liveaction as lv_db_access
+from st2common.persistence import workflow as wf_db_access
+from st2common.services import action as action_service
+from st2common.services import workflows as workflow_service
+from st2common.transport import liveaction as lv_ac_xport
+from st2common.transport import workflow as wf_ex_xport
+from st2common.transport import publishers
+from st2tests.mocks import execution as mock_ac_ex_xport
+from st2tests.mocks import liveaction as mock_lv_ac_xport
+from st2tests.mocks import workflow as mock_wf_ex_xport
+
+TEST_PACK = 'orquesta_tests'
+TEST_PACK_PATH = st2tests.fixturesloader.get_fixtures_packs_base_path() + '/' + TEST_PACK
+
+PACKS = [
+    TEST_PACK_PATH,
+    st2tests.fixturesloader.get_fixtures_packs_base_path() + '/core'
+]
+
+MOCK_NOTIFY = {
+    'on-complete': {
+        'data': {
+            'source_channel': 'baloney',
+            'user': 'lakstorm'
+        },
+        'routes': [
+            'hubot'
+        ]
+    }
+}
+
+
+@mock.patch.object(
+    notifier.Notifier,
+    '_post_notify_triggers',
+    mock.MagicMock(return_value=None))
+@mock.patch.object(
+    notifier.Notifier,
+    '_post_generic_trigger',
+    mock.MagicMock(return_value=None))
+@mock.patch.object(
+    publishers.CUDPublisher,
+    'publish_update',
+    mock.MagicMock(side_effect=mock_ac_ex_xport.MockExecutionPublisher.publish_update))
+@mock.patch.object(
+    lv_ac_xport.LiveActionPublisher,
+    'publish_create',
+    mock.MagicMock(side_effect=mock_lv_ac_xport.MockLiveActionPublisher.publish_create))
+@mock.patch.object(
+    lv_ac_xport.LiveActionPublisher,
+    'publish_state',
+    mock.MagicMock(side_effect=mock_lv_ac_xport.MockLiveActionPublisher.publish_state))
+@mock.patch.object(
+    wf_ex_xport.WorkflowExecutionPublisher,
+    'publish_create',
+    mock.MagicMock(side_effect=mock_wf_ex_xport.MockWorkflowExecutionPublisher.publish_create))
+@mock.patch.object(
+    wf_ex_xport.WorkflowExecutionPublisher,
+    'publish_state',
+    mock.MagicMock(side_effect=mock_wf_ex_xport.MockWorkflowExecutionPublisher.publish_state))
```

Comment:
That's a lot of mocking.

Are all those needed and there is no way around that? (in the past while fixing some test issues I noticed a lot of tests have copied mocks from other places and a lot were actually not needed.

Reply:
@Kami it is the right set of mocking. You should take a closer look at how I write unit tests for orquesta.  i'm only mocking MQ/messaging here. I'm trying NOT to mock any actual functionality. Once a live action is requested, the unit tests will actually run the workflow and the actions to completion and test rather the notification is called or not.